MovieQuiz/Presentation/MovieQuizViewController.swift

```diff
@@ -75,24 +120,38 @@ final class MovieQuizViewController: UIViewController {
             drawBorder(color: .ypRed)
         }
         DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) { [weak self] in
-            self?.showNextQuestionOrResults()
+            guard let self = self else { return }
+            self.showNextQuestionOrResults()
         }
     }
     
     private func showNextQuestionOrResults() {
         hideBorder()
-        if currentQuestionIndex == questions.count - 1 {// - 1 потому что индекс начинается с 0, а длинна массива — с 1
-            let text = "Ваш результат \(correctAnswers) из 10"
-            let viewModel = QuizResultsViewModel(title: "Этот раунд окончен", text: text, buttonText: "Сыграть еще раз")
-            show(quiz: viewModel)
+        if currentQuestionIndex == questionsAmount - 1 {// - 1 потому что индекс начинается с 0, а длинна массива — с 1
+            statisticService?.store(correct: correctAnswers, total: questionsAmount)
+            let gamesCount = statisticService?.gamesCount ?? 0
+            let bestGameCorrect = statisticService?.bestGame.correct ?? 0
+            let bestGameTotal = statisticService?.bestGame.total ?? 0
+            let bestGameDate = statisticService?.bestGame.date.dateTimeString ?? "0"
+            let totalAccuracy = statisticService?.totalAccuracy ?? 0
+            let text = """
+        Ваш результат \(correctAnswers) из 10
+        Количество сыгранных квизов: \(gamesCount)
+        Рекорд: \(bestGameCorrect) / \(bestGameTotal) (\(bestGameDate))
+        Cредняя точность: \(String(format: "%.2f", totalAccuracy * 100))%
+        """
```

Comment:
Автоматическое форматирование не выровняет, к сожалению, но лучше бы выглядело вот так:
```suggestion
            let totalAccuracy = statisticService?.totalAccuracy ?? 0
            let text = """
            Ваш результат \(correctAnswers) из 10
            Количество сыгранных квизов: \(gamesCount)
            Рекорд: \(bestGameCorrect) / \(bestGameTotal) (\(bestGameDate))
            Cредняя точность: \(String(format: "%.2f", totalAccuracy * 100))%
            """
```